runner.py

```diff
@@ -0,0 +1,65 @@
+# Text-based adventure game
+
+inventory, wordle_solution, wordle_attempts = [], "magic", 0
+current_room, game_over = "entrance", False
+
+rooms = {
+    "entrance": {"desc": "You are at the entrance of a mysterious castle.", "exits": {"north": "hall", "east": "library"}, "items": ["key"], "chars": ["guard"]},
+    "hall": {"desc": "A grand hall with statues. There's a locked door to the north.", "exits": {"south": "entrance"}, "items": [], "chars": ["sage"]},
+    "library": {"desc": "A dusty library filled with old books.", "exits": {"west": "entrance"}, "items": ["book"], "chars": []},
+    "treasure_room": {"desc": "The treasure room! Solve the puzzle to claim your prize.", "exits": {"south": "hall"}, "items": [], "chars": []}
+}
+
+characters = {
+    "guard": {"desc": "A stern guard blocks your path.", "interaction": "Guard: 'I can't let you pass unless you give me the key.'"},
+    "sage": {"desc": "An old sage who knows riddles.", "interaction": "Sage: 'Solve the treasure room puzzle to win.'"}
+}
+
+def show_room():
```

Comment:
If you turn the rooms into objects, this method should go inside that class so you could run it on any room at any time!